compiler/src/dotty/tools/dotc/reporting/trace.scala

```diff
@@ -76,7 +76,7 @@ trait TraceSyntax:
     else
       // Avoid evaluating question multiple time, since each evaluation
       // may cause some extra logging output.
-      val q = question.replace('\n', ' ')
+      val q = question
```

Comment:
Could we keep this?  I find having traces be multi-line really disrupts the readability of it.

I realise it impacts the readability of something like TreeTypeMap, but I'm suggesting we bias towards the overall trace.

Reply:
The alternative makes it unreadable. I had to make this change every single time I used trace.

Reply:
Hmm ok